### PR TITLE
Fix typing error in Trainer class (prediction_step)

### PR DESCRIPTION
**Pull request by jannisborn**
# What does this PR do?

This is a minor fix. The argument types and docstring of `transformers.trainer.prediction_step` are incorrect. This error was introduced in transformers==3.4.0, specifically in #7767 where documentation was not updated properly.

The current docs indicate that `prediction_step` returns a 3-Tuple of Optionals (loss, logits and labels) and that the type of _loss_ is `float`. Indeed, if returned, `loss` is always a `torch.Tensor` as the only performed operations in this function are `.mean()`, `.detach()` and `.cpu()`, but **not** `.item()`.  In transformers<3.4.0, there was indeed a `.item()` operation, but in #7767 this behavior was changed but the docstring and types were not updated.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.
@sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!
